### PR TITLE
WT-12820 Allow for comments in the clang_scan.diff file

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: dist/s_clang_scan [hunk 1]
```diff
@@ -34,6 +34,9 @@ $scan $args cmake -G Ninja ../. > /dev/null
 $scan $args ninja wt > ../$t_out 2>&1
 cd ..
 
+# Strip comments from the diff file.
+sed -i '/^#/d' dist/s_clang_scan.diff
```
Comment (ajmorton) [inline dist/s_clang_scan:38]: This is modifying in-place and on each run of `s_clang_scan` we're left with this file stripped of comments. We'll need to save the stripped output to a temp file for comparisons.

**Review by ajmorton — COMMENTED** [review]
I have an alternate proposal for this. If we define the macro
```
#ifdef __clang_analyzer__
#define WT_CLANG_ANALYZE_SUPPRESS __attribute__((suppress))
#else
#define WT_CLANG_ANALYZE_SUPPRESS
#endif
```
we can add suppressions in line with the potential problem code. I've only spent ~15 minutes on this but it looks promising and would allow us to drop the `s_clang_scan.diff` file entirely.

**Comment by luke-pearson** [comment]
> I have an alternate proposal for this. If we define the macro
> 
> ```
> #ifdef __clang_analyzer__
> #define WT_CLANG_ANALYZE_SUPPRESS __attribute__((suppress))
> #else
> #define WT_CLANG_ANALYZE_SUPPRESS
> #endif
> ```
> 
> we can add suppressions in line with the potential problem code. I've only spent !15 minutes on this but it looks promising and would allow us to drop the `s_clang_scan.diff` file entirely.

Okay so if we do it this way we leave a bunch of ifdefs in the code in this PR (with FIXMES aimed at the follow up tickets) and delete the diff file. Eventually the actual investigations will happen and the FIXMEs will be turned into valid comments or the suppression removed and the issue fixed.

I like this method.

File: dist/s_clang_scan (hunk 1 above)
Comment (luke-pearson) [reply dist/s_clang_scan]: [c92ff7c](https://github.com/wiredtiger/wiredtiger/pull/10481/commits/c92ff7c294f63b1248f4d5a2ca48ebd9541bf861)

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Pull request merged**